symplyphysics/laws/chemistry/number_of_collisions_of_atomized_atom_with_gas_atoms_in_magnetron.py

```diff
@@ -0,0 +1,49 @@
+from sympy import Eq, solve, log
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+    validate_output, dimensionless, convert_to_float)
+
+# Description
+## The atoms of the target material evaporate and move towards the substrate inside the magnetron.
+## The atomized atom moves towards the substrate in the magnetron. At the same time, it collides with gas atoms.
+## The number of collisions of a atomized atom, after which its energy will be equal to the energy of thermal motion
+## in a gas-discharge plasma, can be calculated. This amount will depend on the initial energy of the atomized atom and
+## the energy transfer coefficient between the atom and the gas atoms.
+
+
+## Law is: N = ln(Et / E0) / ln(1 - x), where
+## N - the number of collisions of atomized atom with gas atoms (this is a static quantity, it may not be an integer),
+## E0 - initial energy of the atomized atom,
+## Et - energy of thermal motion in a gas-discharge plasma,
+## x - energy transfer coefficient between the atom and the gas atoms.
+
+number_of_collisions_of_atoms = Symbol("number_of_collisions_of_atoms", dimensionless)
+
+initial_energy = Symbol("initial_energy", units.energy)
+energy_of_thermal_motion = Symbol("energy_of_thermal_motion", units.energy)
+energy_transfer_coefficient = Symbol("energy transfer coefficient", dimensionless)
+
+law = Eq(number_of_collisions_of_atoms, log(energy_of_thermal_motion / initial_energy) / log(1 - energy_transfer_coefficient))
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(initial_energy_=initial_energy,
+    energy_of_thermal_motion_=energy_of_thermal_motion,
+    energy_transfer_coefficient_=energy_transfer_coefficient)
+@validate_output(number_of_collisions_of_atoms)
+def calculate_number_of_collisions_of_atoms(initial_energy_: Quantity, energy_of_thermal_motion_: Quantity,
+                                            energy_transfer_coefficient_: float) -> float:
+    if energy_transfer_coefficient_ >= 1:
+        raise ValueError("Energy transfer coefficient must be less than the 1")
+    if initial_energy_.scale_factor <= energy_of_thermal_motion_.scale_factor:
```

Comment:
Why equal is forbidden?

Reply:
> Why equal is forbidden?

It won't make sense. If there is equality, then the substrate on which the target material is sprayed will simply have to be placed on the target. And this is not normal. But from the point of view of a simple formula, equality is possible and logical. Allow equality?

Reply:
Yes, allow it.